opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -199,3 +200,59 @@ def collect(self) -> Optional[Histogram]:
             explicit_bounds=self._boundaries,
             aggregation_temporality=AggregationTemporality.DELTA,
         )
+
+
+def _convert_aggregation_temporality(
+    previous_point: Optional[_PointVarT],
+    current_point: _PointVarT,
+    aggregation_temporality: AggregationTemporality,
+) -> _PointVarT:
```

Comment:
```suggestion
) -> _PointVarT:
    """Converts current_point to the requested aggregation_temporality given the previous_point
    
    previous_point must have CUMULATIVE temporality. current_point may be DELTA or CUMULATIVE.
    The output point will have temporality aggregation_temporality. Since GAUGE points have no
    temporality, they are returned unchanged.
    """
```

Reply:
Curious as to why previous_point must be CUMULATIVE?

Reply:
Also just wondering in what cases would previous point temporality would be different than current point temporality?

Reply:
@aabmass  to comment before resolving this conversation.

Reply:
Hopefully this illustrates how it's intended to work

```
time --->

previous_point_delta:
          |-------|

previous_point_cumulative:
|-----------------|

current_point_cumulative:
|-------------------------|

current_point_delta:
                  |-------|
```

Given you saved `previous_point_cumulative` from the previous collection:
- If you have `current_point_delta` (synchronous instrument), you can generate `current_point_cumulative = previous_point_cumulative + current_point_delta`
- If you have `current_point_cumulative` (async instrument), you can generate `current_point_delta = current_point_cumulative - previous_point_cumulative`

If you saved instead `previous_point_delta`, you wouldn't be able to do either of the above. Check out the [Temporality Conversion section](https://docs.google.com/document/d/1FOmpGCiZAWTd6a3vcEHBnm1LKp7HAKuD4BBN3YCkymc/edit#heading=h.ulghopzfw7ou) of the doc too

Reply:
> Also just wondering in what cases would previous point temporality would be different than current point temporality?

We only keep the previous point as a  CUMULATIVE, regardless of what the reader wants. So if the MetricReader requests DELTA, I would still store the previous point as a CUMULATIVE in order to support the conversions.

Reply:
@aabmass 
Nice, thanks for the explanation.